web-app/js/portal/ui/openlayers/control/SpatialConstraint.js

```diff
@@ -65,11 +67,126 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
         });
     },
 
-    _onSketchStarted: function(e) {
+    _isRegularPolygon: function() {
+        return (this.handler.line === undefined || this.handler.line == null);
+    },
 
-        this.triggerMapClick(e);
-        if (this.map.mapPanel) {
-            this.map.mapPanel._closeFeatureInfoPopup();
+    _polygonCoordsReset: function() {
+        this.polygonCoords = []
+    },
+
+    _addPolygonCoord: function(lonLat) {
+        this.polygonCoords.push(lonLat);
+    },
+
+    _polygonLastCoord: function() {
+        return this.polygonCoords[this.polygonCoords.length-1]
+    },
+
+    _polygonCoordsExist: function() {
+        return this.polygonCoords.length >= 1;
+    },
+
+    _antimeridianPointCross: function(lon1, lon2) {
```

Comment:
could you rename this to __polygonAntimeridianPointCross

Reply:
thanks @pmbohm that makes good sense. I'll wait to see if you have more suggestions and then fix it up